corehq/motech/repeaters/repeater_generators.py

```diff
@@ -448,55 +468,96 @@ def validate(self):
 
         if self.includes is not None and self.excludes is not None:
             raise DataRegistryCaseUpdateError("Both exclude and include lists specified. Only one is allowed.")
-        if self.includes is None and self.excludes is None:
-            raise DataRegistryCaseUpdateError("Neither exclude and include lists specified. One is required.")
+
+    @index_create_relationship.validator
+    @index_remove_relationship.validator
+    def _check_index_relationship(self, attribute, value):
+        if value and value not in ("child", "extension"):
+            raise DataRegistryCaseUpdateError("Index relationships must be either 'child' or 'extension'")
 
     def get_case_block(self, target_case):
+        kwargs = {}
+        if target_case is None:
+            if not self.create_case:
+                # should never get here but added as a precaution
+                raise DataRegistryCaseUpdateError("Target case not found")
+            if not self.owner_id:
+                raise DataRegistryCaseUpdateError("'owner_id' required when creating cases")
+
+            kwargs = {
+                "case_type": self.case_type,
+            }
         return CaseBlock(
-            create=False,
-            case_id=target_case.case_id,
-            update=self.get_case_updates(target_case),
-            index=self.get_case_index(target_case)
+            create=target_case is None,
+            case_id=self.case_id,
```

Comment:
Should case creations generate the ID on the backend, or do you think projects might need access to the ID in the form when creating it?

Reply:
yes exactly. We always generate new case IDs in the forms so that they are are accessible in the form for indexes etc.